components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -33,6 +31,15 @@ class GeckoEngineSession(
         geckoSession.trackingProtectionDelegate = createTrackingProtectionDelegate()
     }
 
+    /**
+     * See [EngineSession.settings]
+     */
+    override val settings: Settings = object : Settings() {
+        override var javascriptEnabled: Boolean
+            get() = runtime.settings.javaScriptEnabled
+            set(value) { runtime.settings.javaScriptEnabled = value }
```

Comment:
I'm still unsure about modifying a runtime setting from a Session. The API and doc make it look like we are enabling/disabling JavaScript just for this session when in fact we are enabling/disabling for all of them.

Reply:
Agreed. This shouldn't be merged as is. The problem is that for GeckoView this is really an engine setting, but for our system engine, it's a session or view setting. We'd want to stay on the side of GV, but if we move this to engine that it won't work at all for WebView which is where this requirement came from. I have thought of some options we should discuss...

Reply:
@csadilek Couldn't we apply the "SystemEngineRuntimeSetting" to every WebView that we create and emulate the same behavior?

@snorp Is JavaScript really only a runtime option in Gecko or could we also expose this as a session-scoped setting?

Reply:
@pocmo yes that's an update I am working on.   Introduce engine settings and pass along to the session (and view). Then the implementations can decide how to handle it.